src/registries/npm-registry.js

```diff
@@ -98,7 +98,7 @@ export default class NpmRegistry extends Registry {
   async getPossibleConfigLocations(filename: string): Promise<Array<[boolean, string, string]>> {
     const possibles = [
       [false, path.join(this.cwd, filename)],
-      [true, this.config.userconfig || path.join(userHome, filename)],
+      [true, this.config.userconfig || path.join(process.env.YARN_HOME_FOLDER || userHome, filename)],
```

Comment:
I have modified the `util-home-dir` to export a function instead. This allows for later extension if developers want to modify their home dir too. I hope that this was what you were looking for, else please let me know!